### PR TITLE
Executor integration

### PR DESCRIPTION
**Pull request by iulianpascalau**
- finished integration of the new state machine
- added semi-integration test for the state machine & the bridge executor with mock bridge clients

### CODE REVIEW

**Review by mihaieremia — DISMISSED** [review]

File: integrationTests/mock/bridgeMock.go [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+func (bm *BridgeMock) SetActionID(actionID bridge.ActionId) {
+	bm.Lock()
+	defer bm.Unlock()
+
+	bm.actionID = actionID
+}
+
+// WasProposedSetStatus -
+func (bm *BridgeMock) WasProposedSetStatus(_ context.Context, batch *bridge.Batch) bool {
+	bm.Lock()
```
Comment (bogdan-rosianu) [inline integrationTests/mock/bridgeMock.go:105]: RLock/RUnlock

File: relay/bridgeStub.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+func (b *bridgeStub) lock() {
+	b.proposeSetStatusMutex.Lock()
+	b.proposeTransferMutex.Lock()
+	b.signMutex.Lock()
+	b.executeMutex.Lock()
+}
+
+// GetPending -
+func (b *bridgeStub) GetPending(_ context.Context) *bridge.Batch {
+	defer func() { b.pendingBatchCallIndex++ }()
```
Comment (bogdan-rosianu) [inline relay/bridgeStub.go:42]: add function body on new line?

Comment (bogdan-rosianu) [reply relay/bridgeStub.go:42]: also, should `pendingBatchCallIndex` be protected?

File: relay/bridgeStub.go [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+import (
+	"context"
+	"sync"
+
+	"github.com/ElrondNetwork/elrond-eth-bridge/bridge"
+)
+
+// TODO remove this after the relay.go full refactoring
+
+type bridgeStub struct {
```
Comment (bogdan-rosianu) [inline relay/bridgeStub.go:12]: this is more like a mock

File: relay/relay.go [hunk 4]
```diff
@@ -150,19 +155,67 @@ func (r *Relay) Start(ctx context.Context) error {
...
 	r.timer.Start()
 
-	monitorEth := NewMonitor(r.ethBridge, r.elrondBridge, r.timer, r, r.quorumProvider, "EthToElrond")
-	go monitorEth.Start(ctx)
-	monitorElrond := NewMonitor(r.elrondBridge, r.ethBridge, r.timer, r, r.quorumProvider, "ElrondToEth")
-	go monitorElrond.Start(ctx)
-
-	<-ctx.Done()
-	if err := r.Stop(); err != nil {
+	smEthToElrond, err := r.createAndStartBridge(r.ethBridge, r.elrondBridge, "EthToElrond")
```
Comment (bogdan-rosianu) [inline relay/relay.go:158]: `ethToElrondStateMachine`?

File: relay/relay.go [hunk 5]
```diff
@@ -150,19 +155,67 @@ func (r *Relay) Start(ctx context.Context) error {
...
+	if err != nil {
+		return nil, err
+	}
+
+	logStateMachine := logger.GetOrCreate(name + "/statemachine")
+	argsStateMachine := stateMachine.ArgsStateMachine{
+		StateMachineName:     name,
+		Steps:                stepsMap,
+		StartStateIdentifier: ethToElrond.GettingPending,
+		DurationBetweenSteps: time.Second * 5,
```
Comment (bogdan-rosianu) [inline relay/relay.go:211]: extract as a const? or TODO to move it in config?

File: integrationTests/mock/bridgeMock.go (hunk 1 above)
Comment (iulianpascalau) [reply integrationTests/mock/bridgeMock.go:105]: done

File: relay/bridgeStub.go (hunk 2 above)
Comment (iulianpascalau) [reply relay/bridgeStub.go:42]: Yes, that was the old code

Comment (iulianpascalau) [reply relay/bridgeStub.go:42]: Fixed the stub

**Review by bogdan-rosianu — APPROVED** [review]

**Pull request merged**